### ISSUE TITLE
Support "WearGo" and "Garmin Smartphone Link" as navigation tools

### ISSUE THREAD

**Issue by gcey** [cgeo/cgeo#12951]
### Tell us your idea!

In "Settings" - "Routing / Navigation" - "Standard navigation" - "Select tool" I can select several tools (e.g. "Locus", "OsmAnd", "WhereYouGo", "Pebble", ...), if these apps are installed on my phone.
Please support also "WearGo" and "Garmin Smartphone Link" in this way.

### Is there an alternative?

It works with "Other external app", but there I have to select "WearGo" or "Garmin Smartphone Link" every time (as I still want to be able to select any app via "Other external app").

**Comment by gcey** [comment]
What do you think about this idea?

**Comment by Lineflyer** [comment]
Do you have any information / documentation about those applications package ID and intents provided for navigation?

**Comment by gcey** [comment]
**WearGo**:
https://play.google.com/store/apps/details?id=**com.devkor.weargo**

**Garmin Smartphone Link**:
https://play.google.com/store/apps/details?id=**com.garmin.android.apps.phonelink**

**Intent**: Both can be used via "Other external app".

Do you need more information?

**Comment by ztNFny** [comment]
Took a short look at WearGo, it offers no intents other than the general navigation ones, so can't be done unless the apps author adds some intents first.

**Comment by moving-bits** [comment]
Wouldn't be the app supporting a nav link sufficient for this? IIRC we need to create an app-specific navigation app in c:geo (see folder `apps/navi`) to get the app listed as navigation target, so that the user can set it as default navigation app.

**Comment by ztNFny** [comment]
> Wouldn't be the app supporting a nav link sufficient for this? IIRC we need to create an app-specific navigation app in c:geo (see folder `apps/navi`) to get the app listed as navigation target, so that the user can set it as default navigation app.

I looked at several and they are all either calling a specific intent (e.g. com.oruxmaps.VIEW_MAP_ONLINE) or a special url (e.g. osmand.api://..) which is handled only by that 1 app.
WearGo currently offers neither a specific intent nor a URL only supported by it afaict from the decompiled apk.

**Comment by Lineflyer** [comment]
> WearGo currently offers neither a specific intent nor a URL only supported by it afaict from the decompiled apk.

For that reason we mainly invented the "Other external app" navigation item. This allows user to choose their preferred app case by case there but still be able to bind the "Navigation (Driving)", etc. to e.g. Google Maps.

**Comment by ztNFny** [comment]
> This allows user to choose their preferred app case by case there but still be able to bind the "Navigation (Driving)", etc. to e.g. Google Maps.

Yes, and? OP is asking for an own shortcut for WearGo instead of the generic "other" - which is currently not possible as WearGo provides no way to address it specifically.

**Comment by fm-sys** [comment]
> which is currently not possible as WearGo provides no way to address it specifically.

It is possible to add a target package parameter to the intent...

**Comment by gcey** [comment]
Do you need any addititional information or anything else, in order to implement this?
